pandas/io/common.py

```diff
@@ -126,9 +127,17 @@ def _validate_header_arg(header) -> None:
         )
 
 
-def _stringify_path(
-    filepath_or_buffer: FilePathOrBuffer[AnyStr]
-) -> FilePathOrBuffer[AnyStr]:
+@overload
+def _stringify_path(filepath_or_buffer: Union[str, Path]) -> str:
+    ...
+
+
+@overload
+def _stringify_path(filepath_or_buffer: IO[AnyStr]) -> IO[AnyStr]:
```

Comment:
i think we will want to add explicit type parameters to all generics as more type hints are added so that we may add `--disallow-any-generics` to allow stricter checking in the future.